crud/common/sharding_key.lua

```diff
@@ -115,6 +116,87 @@ function sharding_key_module.fetch_on_router(space_name)
     return cache.sharding_key_def
 end
 
+-- Make sure sharding key definition is a part of primary key.
+function sharding_key_module.is_part_of_pk(space_name, space_format,
+                                           primary_index, sharding_key_def)
```

Comment:
If it is exported only for unit testing, let's lower it to local and expose as `sharding_key.internal.is_part_of_pk()`.

Reply:
Fixed.

```lua
diff --git a/crud/common/sharding_key.lua b/crud/common/sharding_key.lua
index 5acb5f8..ab00c21 100644
--- a/crud/common/sharding_key.lua
+++ b/crud/common/sharding_key.lua
@@ -119,8 +119,7 @@ function sharding_key_module.update_sharding_keys_cache(space_name)
 end
 
 -- Make sure sharding key definition is a part of primary key.
-function sharding_key_module.is_part_of_pk(space_name, space_format,
-                                           primary_index, sharding_key_def)
+local function is_part_of_pk(space_name, space_format, primary_index, sharding_key_def)
     dev_checks('string', 'table', 'table', 'table')
 
     if cache.as_index_object[space_name] == nil then
@@ -181,8 +180,7 @@ function sharding_key_module.extract_from_pk(space_obj, sharding_key)
     local primary_index = space_obj.index[0]
     local space_format = space_obj:format()
 
-    local is_part_of_pk = sharding_key_module.is_part_of_pk(space_name, space_format,
-                                                            primary_index, sharding_key_def)
+    local is_part_of_pk = is_part_of_pk(space_name, space_format, primary_index, sharding_key_def)
     if is_part_of_pk == false then
         return nil, ShardingKeyError:new(
             "Sharding key for space %q is missed in primary index, specify bucket_id",
@@ -203,4 +201,9 @@ function sharding_key_module.init()
    _G._crud.fetch_on_storage = sharding_key_module.fetch_on_storage
 end
 
+local internal = {
+    is_part_of_pk = is_part_of_pk,
+}
+
+table.insert(sharding_key_module, internal)
 return sharding_key_module
diff --git a/test/unit/sharding_key_test.lua b/test/unit/sharding_key_test.lua
index 8658831..73fe56f 100644
--- a/test/unit/sharding_key_test.lua
+++ b/test/unit/sharding_key_test.lua
@@ -127,8 +127,9 @@ g.test_is_part_of_pk_positive = function()
     }
     local sharding_key_def = {'name', 'age'}
 
-    local res = sharding_key_module.is_part_of_pk(space_name, space_format,
-                                                  {parts = index_parts}, sharding_key_def)
+    local res = sharding_key_module.internal.is_part_of_pk(space_name, space_format,
+                                                           {parts = index_parts},
+                                                           sharding_key_def)
     t.assert_equals(res, true)
 end
 
@@ -145,7 +146,8 @@ g.test_is_part_of_pk_negative = function()
     }
     local sharding_key_def = {'name'}
 
-    local res = sharding_key_module.is_part_of_pk(space_name, space_format,
-                                                  {parts = index_parts}, sharding_key_def)
+    local res = sharding_key_module.internal.is_part_of_pk(space_name, space_format,
+                                                           {parts = index_parts},
+                                                           sharding_key_def)
     t.assert_equals(res, false)
 end
```